src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseDataFileService.cs

```diff
@@ -263,9 +268,9 @@ await ids.Select(id => _releaseFileRepository.CheckFileExists(releaseId, id, Fil
                                             MetaFileId = metaFile.Id,
                                             MetaFileName = metaFile.Filename,
                                             Rows = blob.GetNumberOfRows(),
-                                            UserName = blob.GetUserName(),
+                                            UserName = dataFile.CreatedBy.Email,
```

Comment:
Is CreatedBy always guaranteed to be not null?  I'm just thinking of the case in the actual migration code where you're having to check if the user's email address from the metadata isn't null.  Also I see mention in other places where the `?` operator has been used after CreatedBy

Reply:
It's used with the `?` elsewhere in case there was no CreatedById set e.g. the blob didn't have a `userName` field (yes there are one or two like that!) or we didn't match the email in the `userName` property with a User in the db.

We can guarantee it's not null here because straight above this the file is created with the Id from the user service and then we do an explicit load of `CreatedBy` to ensure its loaded.

The only way for it to be null here I guess would be if the user wasn't authenticated somehow.

Reply:
Cool ta!